src/internals/StreamHttpRequest.hpp

```diff
@@ -370,6 +379,40 @@ const ArduinoHttpServer::ErrorString ArduinoHttpServer::StreamHttpRequest<MAX_BO
    return errorString;
 }
 
+template <size_t MAX_BODY_SIZE>
+bool ArduinoHttpServer::StreamHttpRequest<MAX_BODY_SIZE>::authenticate(const char * username, const char * password) const
+{
+   if (m_authorizationField.getType() == HttpField::Type::NOT_SUPPORTED)
+   {
+      return false;
+   }
+
+   if (!m_authorizationField.getValueAsString().startsWith("Basic"))
+   {
+      DEBUG_ARDUINO_HTTP_SERVER_PRINTLN("Unsupported auth header");
+      return false;
+   }
+
+   String combinedInput;
+   if (!combinedInput.reserve(strlen(username) + strlen(password) + 2))
+   {
+      DEBUG_ARDUINO_HTTP_SERVER_PRINTLN("Not enough memory");
+      return false;
+   }
+   combinedInput += username;
+   combinedInput += AHS_F(":");
+   combinedInput += password;
+
+   int encodedLength = Base64.encodedLength(combinedInput.length());
+   char encodedString[encodedLength];
+   Base64.encode(encodedString, combinedInput.c_str(), combinedInput.length());
+   
+   if (strcmp(authData.c_str()+6,encodedString) == 0)
```

Comment:
Where is this authData coming from?

Reply:
Sorry. I pushed work in progress. fixed.